src/components/DataViewer/DataViewer.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+
+import React from "react";
+import styled from "styled-components";
+import Modal from "react-modal";
+
+interface DataViewerProps {
+    data: { [key: string]: string | number | null };
+    title: string;
+    isOpen: boolean;
+    onRequestClose: () => void;
+}
+
+const ModalWithStyle = styled(Modal)`
+    position: absolute;
+    inset: 50% 40px 40px 50%;
+    border: 0px;
+    background: rgb(255, 255, 255);
+    overflow: auto;
+    border-radius: 5px;
+    outline: none;
+    padding: 0px;
+    width: 80%;
+    height: 80%;
+    box-shadow: rgb(200, 200, 200) 0px 2px 10px;
+    transform: translate(-50%, -50%);
+    overflow: hidden;
+`;
+
+const Header = styled.h3`
+    display: flex;
+    justify-content: space-between;
+    background-color: #eeeeee;
+    margin-bottom: 1rem;
+    margin-top: 0;
+    padding: 0.5rem;
+`;
+
+const Content = styled.div`
+    overflow: scroll;
+    height: calc(100% - 5rem);
+    margin: 1.5rem;
+`;
+
+const Key = styled.div`
+    color: grey;
+    font-size: small;
+`;
+
+const Value = styled.div`
+    font-size: medium;
+`;
+
+const EachItem = styled.div`
+    display: flex;
+    flex-direction: column;
```

Comment:
it should work without display: flex and flex-direction: column as you aren't using any other flexbox properties

Reply:
Agreed